cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/hdfs/HdfsStorageClient.java

```diff
@@ -0,0 +1,59 @@
+package com.linkedin.openhouse.cluster.storage.hdfs;
+
+import com.google.common.base.Preconditions;
+import com.linkedin.openhouse.cluster.storage.StorageClient;
+import com.linkedin.openhouse.cluster.storage.StorageType;
+import com.linkedin.openhouse.cluster.storage.configs.StorageProperties;
+import java.io.IOException;
+import javax.annotation.PostConstruct;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hadoop.fs.FileSystem;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+/**
+ * The HdfsStorageClient class is an implementation of the StorageClient interface for HDFS storage.
+ * It uses an Apache Hadoop FileSystem to interact with the HDFS file system.
+ */
+@Slf4j
+@Lazy
+@Component
+public class HdfsStorageClient implements StorageClient<FileSystem> {
+
+  private FileSystem fs;
+
+  @Autowired private StorageProperties storageProperties;
+
+  private static final StorageType.Type HDFS_TYPE = StorageType.HDFS;
+
+  /** Initialize the HdfsStorageClient when the bean is accessed for the first time. */
+  @PostConstruct
+  public synchronized void init() throws IOException {
+    log.info("Initializing storage client for type: " + HDFS_TYPE);
+    Preconditions.checkArgument(
+        !(storageProperties.getTypes() == null
+            || storageProperties.getTypes().isEmpty()
+            || !storageProperties.getTypes().containsKey(HDFS_TYPE.getValue())),
+        "Storage properties doesn't contain type: " + HDFS_TYPE.getValue());
+    StorageProperties.StorageTypeProperties hdfsStorageProperty =
+        storageProperties.getTypes().get(HDFS_TYPE.getValue());
+    Preconditions.checkArgument(
+        hdfsStorageProperty != null,
+        "Storage properties doesn't contain type: " + HDFS_TYPE.getValue());
+    Preconditions.checkArgument(
+        hdfsStorageProperty.getEndpoint() != null,
+        "Storage properties doesn't contain endpoint for: " + HDFS_TYPE.getValue());
+    Preconditions.checkArgument(
+        hdfsStorageProperty.getRootPath() != null,
+        "Storage properties doesn't contain rootpath for: " + HDFS_TYPE.getValue());
+    org.apache.hadoop.conf.Configuration configuration = new org.apache.hadoop.conf.Configuration();
+    configuration.set("fs.defaultFS", hdfsStorageProperty.getEndpoint());
```

Comment:
How about adding additional resources such as hadoop core site and hdfs site path as part of configuration resources?

Reply:
I looked into our docker setup and deployments, we do not use core-site, hdfs-site. Do you think we should still provision for it here?.. its possible through configurable properties.

Reply:
Yes we have not used them through configs. Like you mentioned we can make them available through configurable properties.

Reply:
do you think we should still pursue this, or wait for the feature request?

Reply:
I think we should be good for now. Good to keep a note of this for future.